src/Fixer/PhpUnit/PhpUnitDataProviderReturnTypeFixer.php

```diff
@@ -0,0 +1,111 @@
+<?php
+
+declare(strict_types=1);
+
+/*
+ * This file is part of PHP CS Fixer.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *     Dariusz Rumiński <dariusz.ruminski@gmail.com>
+ *
+ * This source file is subject to the MIT license that is bundled
+ * with this source code in the file LICENSE.
+ */
+
+namespace PhpCsFixer\Fixer\PhpUnit;
+
+use PhpCsFixer\Fixer\AbstractPhpUnitFixer;
+use PhpCsFixer\FixerDefinition\CodeSample;
+use PhpCsFixer\FixerDefinition\FixerDefinition;
+use PhpCsFixer\FixerDefinition\FixerDefinitionInterface;
+use PhpCsFixer\Tokenizer\Analyzer\DataProviderAnalyzer;
+use PhpCsFixer\Tokenizer\Analyzer\FunctionsAnalyzer;
+use PhpCsFixer\Tokenizer\CT;
+use PhpCsFixer\Tokenizer\Token;
+use PhpCsFixer\Tokenizer\Tokens;
+
+/**
+ * @author Kuba Werłos <werlos@gmail.com>
```

Comment:
```suggestion
 * Fixer for data provider return time matching official expectation described in https://docs.phpunit.de/en/10.2/writing-tests-for-phpunit.html#data-providers : "It must return [iterable](https://www.php.net/manual/en/language.types.iterable.php), either an array of arrays or an object that implements the Traversable interface.".
 * Technically, this update was introduced to documentation of PHPUnit 10, yet it's technically possible already starting in PHPUnit 6.
 *
 * @author Kuba Werłos <werlos@gmail.com>
```

we do something similar for PSR rules, eg
https://github.com/PHP-CS-Fixer/PHP-CS-Fixer/blob/master/src/Fixer/Whitespace/LineEndingFixer.php#L26-L31

Reply:
maybe it could partially be added also to `FixerDefinition` (non-blocking idea)

Reply:
> Technically, this update was introduced to documentation

I'm not sure that looking at documentation over the actual implementation is a good DX. Even looking at the discussion here suggests we are not clear on what direction to follow.

Reply:
Even in PHPUnit 4 all what code needed from data provider is to be able to iterate over it: https://github.com/sebastianbergmann/phpunit/blob/4.8.36/src/Util/Test.php#L374-L401